custom/icds/models.py

```diff
@@ -172,9 +180,18 @@ def update_status(self, new_status):
         HostedCCZ.objects.filter(id=self.pk).update(status=new_status)
 
 
+class HostedCCZCustomSupportingFile(models.Model):
+    link = models.ForeignKey(HostedCCZLink, on_delete=models.PROTECT)
+    file = models.ForeignKey(HostedCCZSupportingFile, on_delete=models.CASCADE)
```

Comment:
delete any corresponding custom supporting file links, if the file itself is getting removed.